nuxeo-runtime/nuxeo-runtime-migration/src/main/java/org/nuxeo/runtime/migration/MigrationServiceImpl.java

```diff
@@ -145,7 +145,39 @@ public void receivedMessage(MigrationInvalidation message) {
                 log.error("Unknown migration id received in invalidation: {}", id);
                 return;
             }
+            // don't send again invalidation, call sub migrator directly
+            ((InvalidatorMigrator) migrator).migrator.notifyStatusChange();
+        }
+    }
+
+    public static class InvalidatorMigrator implements Migrator {
```

Comment:
Please add a small Javadoc explaining that this wraps the migrator to also send invalidations.